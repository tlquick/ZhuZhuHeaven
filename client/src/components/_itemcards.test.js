import { render, screen, cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";
import ItemCards from "./_itemcards";

afterEach(() => {
  cleanup();
});
test("renders itemcards page", async () => {
  render(<ItemCards items={mockedData} />);
  const element = screen.getByTestId("itemcards-1");
  expect(element).toBeInTheDocument();
  expect(element).toContainHTML("div"); //contains a <div> tag
  const items = screen.getAllByTestId("itemcards-2");
  expect(items.length).toBe(3);
});
test("matches itemcards snapshot", () => {
  const tree = renderer.create(<ItemCards items={mockedData} />).toJSON();
  expect(tree).toMatchSnapshot();
});
const mockedData = [
  {
    _id: 1,
    series: 1,
    name: "mr squiggles",
    desc: "Mr Squiggles is brown and white with a pink nose",
    price: 220,
    image: "../image/mrsquiggles.jpg",
  },
  {
    _id: 2,
    series: 1,
    name: "pipsqueak",
    desc: "Pipsqueak is yellow and white with a pink nose",
    price: 200,
    image: "../image/pipsqueak.jpg",
  },
  {
    _id: 3,
    series: 2,
    name: "jilly",
    desc: "Jilly is pink and white with a pink nose",
    price: 200,
    image: "../image/jilly.jpg",
  },
];
