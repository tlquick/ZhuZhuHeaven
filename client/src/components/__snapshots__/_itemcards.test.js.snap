// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches itemcards snapshot 1`] = `
<div
  className="row"
  data-testid="itemcards-1"
>
  <div
    className="col-md-4 mb-4"
  >
    <div
      className="card"
      data-testid="itemcards-2"
    >
      <img
        alt="item"
        className="card-img-top"
        src="../image/mrsquiggles.jpg"
      />
      <div
        className="card-body"
      >
        <h5
          className="card-title"
        >
          mr squiggles
        </h5>
        <p
          className="card-text"
        >
          Mr Squiggles is brown and white with a pink nose
        </p>
        <p>
          Price: $
          220
        </p>
        <a
          className="btn btn-dark"
          href="/cart"
        >
          Add to Cart
        </a>
      </div>
    </div>
  </div>
  <div
    className="col-md-4 mb-4"
  >
    <div
      className="card"
      data-testid="itemcards-2"
    >
      <img
        alt="item"
        className="card-img-top"
        src="../image/pipsqueak.jpg"
      />
      <div
        className="card-body"
      >
        <h5
          className="card-title"
        >
          pipsqueak
        </h5>
        <p
          className="card-text"
        >
          Pipsqueak is yellow and white with a pink nose
        </p>
        <p>
          Price: $
          200
        </p>
        <a
          className="btn btn-dark"
          href="/cart"
        >
          Add to Cart
        </a>
      </div>
    </div>
  </div>
  <div
    className="col-md-4 mb-4"
  >
    <div
      className="card"
      data-testid="itemcards-2"
    >
      <img
        alt="item"
        className="card-img-top"
        src="../image/jilly.jpg"
      />
      <div
        className="card-body"
      >
        <h5
          className="card-title"
        >
          jilly
        </h5>
        <p
          className="card-text"
        >
          Jilly is pink and white with a pink nose
        </p>
        <p>
          Price: $
          200
        </p>
        <a
          className="btn btn-dark"
          href="/cart"
        >
          Add to Cart
        </a>
      </div>
    </div>
  </div>
</div>
`;
